test/matchers/log_format.go

```diff
@@ -0,0 +1,139 @@
+package matchers
+
+import (
+	"fmt"
+	logger "github.com/ViaQ/logerr/log"
+	"github.com/onsi/gomega/types"
+	. "github.com/openshift/cluster-logging-operator/test/helpers"
+	"reflect"
+	"strings"
+)
+
+type LogMatcher struct {
+	expected interface{}
+}
+
+func FitLogFormatTemplate(expected interface{}) types.GomegaMatcher {
+	return &LogMatcher{
+		expected: expected,
+	}
+}
+
+func (m *LogMatcher) Match(actual interface{}) (success bool, err error) {
+	template, ok := m.expected.(Log)
+	if !ok {
+		return false, fmt.Errorf("matcher expects Log")
+	}
+	log, ok := actual.(Log)
+	if !ok {
+		return false, fmt.Errorf("matcher expects Log")
+	}
+
+	return CompareLog(template,log)
+}
+
+func (m *LogMatcher) FailureMessage(actual interface{}) (message string) {
+	return fmt.Sprintf("Expected\n\t%#v\nto fit \n\t%#v", actual, m.expected)
+}
+
+func (m *LogMatcher) NegatedFailureMessage(actual interface{}) (message string) {
+	return fmt.Sprintf("Expected\n\t%#v\nto not fit \n\t%#v", actual, m.expected)
+}
+
+func isNil(i interface{}) bool {
+	if i == nil {
+		return true
+	}
+	switch reflect.TypeOf(i).Kind() {
+	case reflect.Ptr, reflect.Map, reflect.Array, reflect.Chan, reflect.Slice:
+		return reflect.ValueOf(i).IsNil()
+	}
+	return false
+}
+
+func DeepFields(iface interface{}, namePrefix string) ([]reflect.Value, []string) {
+	fields := make([]reflect.Value, 0)
+	names :=  make([]string,0)
+
+	ifv := reflect.ValueOf(iface)
+	ift := reflect.TypeOf(iface)
+
+	for i := 0; i < ift.NumField(); i++ {
```

Comment:
can you not use a range here:
```
for i, name := range ift.NumField
```

Reply:
Nope ...  NumField returns a struct type's field count. (will not work as a range)